src/assertions/OrderAssertions.ts

```diff
@@ -125,4 +131,39 @@ export class OrderAssertions {
       await this.erc20Assertions.implementsERC20(relayerToken);
     }
   }
+
+  public isValidZeroExOrderFills (
+    signedIssuanceOrder: SignedIssuanceOrder,
+    quantityToFill: BigNumber,
+    orders: ExchangeOrder[],
+  ) {
+    let zeroExFillAmounts = SetProtocolUtils.CONSTANTS.ZERO;
+    _.each(orders, (order: any) => {
+      if (SetProtocolUtils.isZeroExOrder(order)) {
+        this.commonAssertions.greaterThanZero(
+          order.fillAmount,
+          coreAPIErrors.QUANTITY_NEEDS_TO_BE_POSITIVE(order.fillAmount),
+        );
+        zeroExFillAmounts = zeroExFillAmounts.plus(order.fillAmount);
+        this.commonAssertions.isEqualString(
+          signedIssuanceOrder.makerToken,
+          SetProtocolUtils.extractAddressFromAssetData(order.takerAssetData),
+          coreAPIErrors.ISSUANCE_ORDER_MAKER_ZERO_EX_TAKER_MISMATCH(),
+        );
+      }
+    });
+
+    // All 0x signed fill order fillAmounts are filled using the makerTokenAmount of the
+    // signedIssuanceOrder so we need to make sure that signedIssuanceOrder.makerTokenAmount
+    // has enough for the 0x orders (scaled by fraction of order quantity being filled).
+    const {
+      makerTokenAmount,
+      quantity,
+    } = signedIssuanceOrder;
+    this.commonAssertions.isGreaterOrEqualThan(
+      signedIssuanceOrder.makerTokenAmount.mul(quantityToFill).div(quantity),
```

Comment:
I believe this assertion needs to hold true for ALL orders. 

So the `makerTokenAmount` needs to exceed both TakerWallet and 0x order usage. So this check needs to be in a level that captures both TakerWallet and 0x.

Reply:
Do I add in the takerTokenAmounts?

Reply:
TakerWalletOrders don't have any maker tokens though?

Reply:
@asoong ^